pkg/ccl/changefeedccl/alter_changefeed_test.go

```diff
@@ -423,11 +422,34 @@ func TestAlterChangefeedTelemetry(t *testing.T) {
 
 		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo, bar WITH diff`)
 		defer closeFeed(t, testFeed)
-
 		feed := testFeed.(cdctest.EnterpriseTestFeed)
 
 		require.NoError(t, feed.Pause())
-		jobutils.WaitForJobToHaveNoLease(t, sqlDB, feed.JobID())
+
+		// The job system clears the lease asyncronously after
+		// cancellation. This lease clearing transaction can
+		// cause a restart in the alter changefeed
+		// transaction, which will lead to different feature
+		// counter counts. Thus, we want to wait for the lease
+		// clear. However, the lease clear isn't guaranteed to
+		// happen, so we only wait a few seconds for it.
+		waitForNoLease := func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Should the period be increased if under race?
helpers_test.go has withTimeout function that can help with this, plus assertPayloadsTimeout() is a helper that returns timeout adjusted for race (might need renaming)